store/kv/fdb.go

```diff
@@ -481,15 +510,16 @@ func (t *ftx) Commit(ctx context.Context) error {
 		log.Err(err).Msg("tx Commit")
 
 		var ep fdb.Error
-		if xerrors.As(err, &ep) {
-			if ep.Code == 1020 {
+		if errors.As(err, &ep) {
+			err = t.tx.OnError(ep).Get()
+
+			if err != nil && ep.Code == 1020 {
 				err = ErrConflictingTransaction
-			} else if err1 := t.tx.OnError(ep).Get(); err1 != nil {
-				err = err1
 			}
 		}
 
 		if err != nil {
+			t.tx.Cancel()
```

Comment:
do we need this cancel in fdbkv as well?

Reply:
Yes, please check `txWithRetry` it has `defer tr.Cancel()`

Reply:
ah, right.